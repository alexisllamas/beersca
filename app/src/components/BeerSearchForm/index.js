import React, { PureComponent } from 'react';
import Input from '../Input';

class BeerSearchForm extends PureComponent {
  state = {
    timer: null,
  };
  handleChange = ev => {
    if (!this.state.timer) {
      return this.createTimer(ev.target.value);
    }
    clearTimeout(this.state.timer);
    this.createTimer(ev.target.value);
  };

  createTimer = query => {
    const timer = setTimeout(() => {
      this.search(query);
    }, 1000);
    this.setState({ timer });
  };

  search = query => {
    const { fetchBeers, searchBeers } = this.props;

    if (!query) return fetchBeers && fetchBeers();

    searchBeers && searchBeers(query);
  };

  render() {
    return (
      <form>
        <Input
          labelText="Beer name"
          className="input"
          name="beerName"
          type="text"
          placeholder="Text input"
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

export default BeerSearchForm;
