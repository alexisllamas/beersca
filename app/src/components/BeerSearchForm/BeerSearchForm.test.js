import React from 'react';
import mount from 'enzyme/mount';
import BeerSearchForm from '.';

describe('BeerSearchForm component', () => {
  const searchBeers = jest.fn();
  const fetchBeers = jest.fn();

  const wrapper = mount(
    <BeerSearchForm searchBeers={searchBeers} fetchBeers={fetchBeers} />,
  );
  it('should have render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call searchBeers', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'Cerveza de Alexis' } });
    setTimeout(() => {
      expect(searchBeers).toBeCalledWith('Cerveza de Alexis');
    }, 1001);
  });
  it('should call fetchBeers when input empty', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: '' } });
    setTimeout(() => {
      expect(searchBeers).toBeCalled(fetchBeers);
    }, 1001);
  });
});
