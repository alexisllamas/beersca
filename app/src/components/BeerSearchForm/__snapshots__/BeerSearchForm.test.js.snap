// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerSearchForm component should have render correctly 1`] = `
<BeerSearchForm
  fetchBeers={[Function]}
  searchBeers={[Function]}
>
  <form>
    <Input
      className="input"
      labelText="Beer name"
      name="beerName"
      onChange={[Function]}
      placeholder="Text input"
      type="text"
    >
      <div
        className="field"
      >
        <label
          className="label"
          htmlFor="beerName"
        >
          Beer name
        </label>
        <div
          className="control"
        >
          <input
            className="input"
            id="beerName"
            name="beerName"
            onChange={[Function]}
            placeholder="Text input"
            type="text"
          />
        </div>
      </div>
    </Input>
  </form>
</BeerSearchForm>
`;
